src/raftstore/store/snap.rs

```diff
@@ -61,10 +66,120 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+pub enum SendSnapshotFileReader {
```

Comment:
For Writer, I think you can use `Box<Writer>`, so you can create the concrete Writer when parsing the first snapshot message. 

For applying snapshot, why not abstract the apply function to trait too?